scene/2d/collision_shape_2d.cpp

```diff
@@ -56,30 +96,26 @@ Color CollisionShape2D::_get_default_debug_color() const {
 
 void CollisionShape2D::_notification(int p_what) {
 	switch (p_what) {
-		case NOTIFICATION_PARENTED: {
-			collision_object = Object::cast_to<CollisionObject2D>(get_parent());
+		case NOTIFICATION_ENTER_TREE: {
+			collision_object = _get_ancestor_collision_object();
 			if (collision_object) {
 				owner_id = collision_object->create_shape_owner(this);
 				if (shape.is_valid()) {
 					collision_object->shape_owner_add_shape(owner_id, shape);
 				}
```

Comment:
*This change* is the main reason I am worried about this change. I'll explain more in the comment, but it affects whether shapes can be created before entering the tree.